### PR TITLE
Cal 242 read interviewers

### PR DESCRIPTION
**Pull request by Annapooraniqxf2**

### CODE REVIEW

File: qxf2_scheduler/routes.py [hunk 1]
```diff
@@ -65,73 +68,89 @@ def index():
 
 
 @app.route("/interviewers")
-def listinterviewer():
-    "List the interviewer names,designation"
-    interviewer_work_time_slots = []
-    new_slot = Interviewers.query.join(Interviewertimeslots,Interviewers.interviewer_id==Interviewertimeslots.interviewer_id).values(Interviewers.interviewer_id,Interviewers.interviewer_email,Interviewertimeslots.interviewer_start_time,Interviewertimeslots.interviewer_end_time)
+def listinterviewers():
```
Comment (qxf2) [inline qxf2_scheduler/routes.py]: Can you please change the method name to be list_interviewers()

File: qxf2_scheduler/routes.py [hunk 2]
```diff
@@ -65,73 +68,89 @@ def index():
...
-    for interviewer_id,interviewer_email,interviewer_start_time,interviewer_end_time in new_slot:        
-        interviewer_details = {'interviewer_id':interviewer_id,'interviewer_email':interviewer_email,'interviewer_start_time':interviewer_start_time,
-        'interviewer_end_time':interviewer_end_time}        
-        if not interviewer_work_time_slots:            
-            interviewer_work_time_slots.append(interviewer_details)
+@app.route("/<interviewer_id>/interviewer/")
+def read_interviewer_details(interviewer_id):
+    "Displays all the interviewer details"
+    parsed_interviewer_details = []
+    interviewer_details = Interviewers.query.join(Interviewertimeslots, Interviewers.interviewer_id == Interviewertimeslots.interviewer_id).filter(
```
Comment (qxf2) [inline qxf2_scheduler/routes.py]: When writing long and complex queries like this, it is a good habit to comment what is going on using English. You or the next developer working on this a year from now will thank you.

File: qxf2_scheduler/routes.py [hunk 3]
```diff
@@ -65,73 +68,89 @@ def index():
...
+
+    return render_template("list-interviewers.html", result=my_interviewers_list)
+
+
+def parse_interviewer_detail(interviewer_details):
+    "Parse the interviewer detail with start and end time"    
+    time_dict = {}
+    time_dict['starttime'] = interviewer_details['interviewers_starttime']
+    time_dict['endtime'] = interviewer_details['interviewers_endtime']   
+    del interviewer_details['interviewers_starttime']
```
Comment (qxf2) [inline qxf2_scheduler/routes.py]: Why are we deleting this here? Also, if you just want to process two pieces of info, couldn't you just pass those two pieces?

File: qxf2_scheduler/routes.py [hunk 4]
```diff
@@ -65,73 +68,89 @@ def index():
...
+    all_interviewers = Interviewers.query.all()
+    my_interviewers_list = []
+    for each_interviewer in all_interviewers:
+        my_interviewers_list.append({'interviewer_id': each_interviewer.interviewer_id,
+                                     'interviewer_name': each_interviewer.interviewer_name})
+
+    return render_template("list-interviewers.html", result=my_interviewers_list)
+
+
+def parse_interviewer_detail(interviewer_details):
```
Comment (qxf2) [inline qxf2_scheduler/routes.py]: Rename the method to be `form_interviewer_timeslot(start_time,end_time)`

File: qxf2_scheduler/routes.py [hunk 5]
```diff
@@ -65,73 +68,89 @@ def index():
...
+    "Displays all the interviewer details"
+    parsed_interviewer_details = []
+    interviewer_details = Interviewers.query.join(Interviewertimeslots, Interviewers.interviewer_id == Interviewertimeslots.interviewer_id).filter(
+        Interviewers.interviewer_id == interviewer_id).values(Interviewers.interviewer_name, Interviewers.interviewer_email, Interviewers.interviewer_designation, Interviewers.interviewer_id, Interviewertimeslots.interviewer_start_time, Interviewertimeslots.interviewer_end_time)
+    for each_detail in interviewer_details:
+        interviewer_detail = {
+            'interviewers_name': each_detail.interviewer_name,
+            'interviewers_email': each_detail.interviewer_email,
+            'interviewers_designation': each_detail.interviewer_designation,
+            'interviewers_starttime': each_detail.interviewer_start_time,
```
Comment (qxf2) [inline qxf2_scheduler/routes.py]: You could replace this as `'time' : parse_interviewer_detail(each_detail.interviewer_start, each_detail.interviewer_end)`

File: qxf2_scheduler/routes.py [hunk 6]
```diff
@@ -65,73 +68,89 @@ def index():
...
+            'interviewers_endtime': each_detail.interviewer_end_time}
+        
+        parsed_interviewer_detail = parse_interviewer_detail(interviewer_detail)
+        if not parsed_interviewer_details:
```
Comment (qxf2) [inline qxf2_scheduler/routes.py]: For lists, it is clearer to check Len(list_name) == 0

File: qxf2_scheduler/routes.py [hunk 7]
```diff
@@ -65,73 +68,89 @@ def index():
...
-                        each_interviewer['interviewer_end_time'] = [each_interviewer['interviewer_end_time'], interviewer_details['interviewer_end_time']]
-                     
-                    break
-             
-            if append_flag == True:                
-                interviewer_work_time_slots.append(interviewer_details)
-               
-    return render_template("list-interviewer.html", result=interviewer_work_time_slots)   
-    
+            if interviewer_detail['interviewers_name'] in parsed_interviewer_details[0].values():
```
Comment (qxf2) [inline qxf2_scheduler/routes.py]: I could not follow the logic here. It looks like this code is checking only the first appended dictionary's values for the name. Shouldn't you be looping through every dictionary in parsed_interviewer_details?

File: qxf2_scheduler/routes.py (hunk 1 above)
Comment (Annapooraniqxf2) [reply qxf2_scheduler/routes.py]: Changed the method name

File: qxf2_scheduler/routes.py (hunk 2 above)
Comment (Annapooraniqxf2) [reply qxf2_scheduler/routes.py]: Done

File: qxf2_scheduler/routes.py (hunk 4 above)
Comment (Annapooraniqxf2) [reply qxf2_scheduler/routes.py]: Done

File: qxf2_scheduler/routes.py (hunk 5 above)
Comment (Annapooraniqxf2) [reply qxf2_scheduler/routes.py]: Done

File: qxf2_scheduler/routes.py (hunk 6 above)
Comment (Annapooraniqxf2) [reply qxf2_scheduler/routes.py]: Done

File: qxf2_scheduler/routes.py (hunk 7 above)
Comment (Annapooraniqxf2) [reply qxf2_scheduler/routes.py]: I am adding answer in the JIRA ticket

**Pull request merged**